### PR TITLE
Add ipv6 support to the e2e kubectl tests

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:

It adds IPv6 support to the `\[sig-cli\] Kubectl client \[k8s.io\] Simple pod"` tests

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Nginx image only listens on IPv4 by default, but Httpd image listens on both IPv4 and IPv6
This is the same we did in https://github.com/kubernetes/kubernetes/pull/78871

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by fejta** [comment]
Can we get someone on @kubernetes/sig-network-pr-reviews to review?
/approve

**Comment by danwinship** [comment]
So @aojea, this basically doesn't do anything except switch from nginx to apache, right? And it's just that, as a side effect, *if* the cluster is configured to do IPv6 single-stack rather than IPv4 single-stack, then the test will end up using IPv6 and pass now, rather than trying to do IPv4 and failing? That all seems reasonable.

**Comment by aojea** [comment]
> So @aojea, this basically doesn't do anything except switch from nginx to apache, right? And it's just that, as a side effect, _if_ the cluster is configured to do IPv6 single-stack rather than IPv4 single-stack, then the test will end up using IPv6 and pass now, rather than trying to do IPv4 and failing? That all seems reasonable.

exactly, upstream httpd image listens on both IPv4 and IPv6 by default, so it works for both families without any change. The problem with nginx image is that it only listens on IPv4 by default, and upstream doesn´t want to change that. I´ve tried to add a custom nginx image listening on both protocols but the preference is to not having a custom image.

File: test/e2e/kubectl/kubectl.go [hunk 1]
```diff
@@ -645,7 +645,7 @@ var _ = SIGDescribe("Kubectl client", func() {
 
 			inClusterHost := strings.TrimSpace(framework.RunHostCmdOrDie(ns, simplePodName, "printenv KUBERNETES_SERVICE_HOST"))
 			inClusterPort := strings.TrimSpace(framework.RunHostCmdOrDie(ns, simplePodName, "printenv KUBERNETES_SERVICE_PORT"))
-
+			inClusterURL := net.JoinHostPort(inClusterHost, inClusterPort)
```
Comment (bowei) [inline test/e2e/kubectl/kubectl.go]: move this down right before use

**Review by bowei — COMMENTED** [review]
looks ok 

/lgtm

**Pull request merged**